proxy/src/telemetry/metrics/prometheus.rs

```diff
@@ -430,11 +430,11 @@ impl fmt::Display for OutboundLabels {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match *self {
             OutboundLabels { namespace: Some(ref ns), dst: Some(ref dst) } =>
-                 write!(f, "dst_namespace=\"{}\",dst_{},", ns, dst),
+                 write!(f, ",dst_namespace=\"{}\",dst_{},", ns, dst),
```

Comment:
does this imply that `dst_namespace` should never be written as the first label, because it's preceded by a comma?

to put it another way, it seems like the safest thing to do is to collect the total set of labels, and do join(",") on them, rather than rely on some kind of implicit ordering.

Reply:
Theoretically, yeah, you're definitely right, and if I were writing a general-purpose Prometheus client library, that's definitely what I'd do. Here, though, we always know the set of labels on a given metric, so the values are stored in a struct rather than a collection, and this way, we can write the formatted values directly to the `Formatter` (i.e. whatever we're outputting to), rather than having to allocate a vector, put all the label value strings in it, format the labels and values values by allocating more strings, join them with commas (allocating another string) and _then_ write them to the `Formatter`, so this is much more efficient. 

I'll document the potential gotchas relating to comma-ordering in a comment.

Reply:
I actually feel it makes sense for the `Display` implementations to not be so aware of where they are being used. I don't think `OutboundLabels` should know it needs to start with a comma, since if we change the order at some point, then we have a leading comma.

It seems probably better to check something like `outbound_labels.is_empty()` from `RequestLabels`, and add a comma if not.

Reply:
@seanmonstar sure, that would also work.

Reply:
@seanmonstar WDYT of a3ae247056c6364b2a484b79421d3bedfd4f4253?